components/service/glean/src/main/java/mozilla/components/service/glean/StringMetricType.kt

```diff
@@ -0,0 +1,59 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.service.glean
+
+import mozilla.components.service.glean.storages.StringsStorageEngine
+import mozilla.components.support.base.log.logger.Logger
+
+/**
+ * This implements the developer facing API for recording string metrics.
+ *
+ * Instances of this class type are automatically generated by the parsers at build time,
+ * allowing developers to record values that were previously registered in the metrics.yaml file.
+ *
+ * The string API only exposes the [record] method, which takes care of validating the input
+ * data and making sure that limits are enforced.
+ */
+data class StringMetricType(
+    override val applicationProperty: Boolean = false,
+    override val disabled: Boolean = false,
+    override val category: String,
+    override val name: String,
+    override val sendInPings: List<String> = listOf("default"),
+    override val userProperty: Boolean = false,
+    val maxLength: Int
```

Comment:
Mh, are we supposed to let users define a limit here or should we settle for one ourselves? This makes sense, but I wonder if that's what we intend to do. I think the spec is not clear wrt this.

Reply:
It looks like the spec had this at one point, but it looks like @georgf removed it on 2018-09-23.  I guess the question is whether we should just have a fixed global limit or we want the complexity of enforcing it from a configurable variable.

Reply:
I would be for removing it: otherwise users of the API might be tempted to use unreasonable limits to force the collection of some specific strings. Let's define a limit ourselves: we can start by using the one from [Desktop scalars](https://searchfox.org/mozilla-central/rev/e0c879c86b95bdc752b1dbff6088169735674e4a/toolkit/components/telemetry/core/TelemetryScalar.cpp#104), `50`. We can always revisit later.

Reply:
Ok.  Going with that...